example/example.py

```diff
@@ -0,0 +1,119 @@
+    # to demonstrate merging coverage from multiple regressions/tests.
+    sampler_a = MySampler(coverage=cvg_a)
+    for _ in range(100):
+        sampler_a.sample(sampler_a.create_trace())
+
+    sampler_b = MySampler(coverage=cvg_b)
+    for _ in range(500):
+        sampler_b.sample(sampler_b.create_trace())
+
+    # Create a context specific hash
+    # This is stored alongside recorded coverage and is used to determine if
+    # coverage is valid to merge.
+    context_hash = Repo().head.object.hexsha
+
+    # Create a reader
+    point_reader = PointReader(context_hash)
+
+    # Read the two sets of coverage
+    reading_a = point_reader.read(cvg_a)
+    reading_b = point_reader.read(cvg_b)
+
+    # Create a local sql database
+    sql_accessor = SQLAccessor.File(db_path)
+
+    # Write each reading into the database
+    rec_ref_a = sql_accessor.write(reading_a)
+    rec_ref_b = sql_accessor.write(reading_b)
+
+    # Read back from sql
+    sql_reading_a = sql_accessor.read(rec_ref_a)
+    sql_reading_b = sql_accessor.read(rec_ref_b)
+
+    # Merge together
+    merged_reading = MergeReading(sql_reading_a, sql_reading_b)
+
+    # Write merged coverage into the database
+    rec_ref_merged = sql_accessor.write(merged_reading)
+
+    # Output to console
+    print("\n-------------------------------------------------------")
+    print("This is the coverage with 100 samples:")
+    print(
+        f"To view this coverage in detail please run: python -m bucket read --sql-path example_file_store --points --record {rec_ref_a}"
+    )
+    ConsoleWriter(axes=False, goals=False, points=False).write(reading_a)
+    print(
+        "\nThis is the coverage from 2 regressions. One with 100 samples, and one with 500:"
+    )
+    print(
+        f"To view this coverage in detail please run: python -m bucket read --sql-path example_file_store --points --record {rec_ref_merged}"
+    )
+    ConsoleWriter(axes=False, goals=False, points=False).write(merged_reading)
+
+    # Read all back from sql - note as the db is not removed this will
+    # accumulate each time this example is run. This will also include
+    # merged data as well as the individual runs. It is meant as an example
+    # of how to use the command
+    merged_reading_all = MergeReading(*sql_accessor.read_all())
+    print("\nThis is the coverage from all the regression data so far:")
+    print("(To reset please delete the file 'example_file_store')")
+    ConsoleWriter(axes=False, goals=False, points=False).write(merged_reading_all)
+
+    # Generating web viewer
+    # To generate the HTML report run:
+    # python -m bucket write html --sql-path ./example_file_store.db --output index.html
+    HTMLWriter(
+        web_path=Path(__file__).parent.parent / "viewer", output="index.html"
+    ).write(merged_reading_all)
+
+    print("\n-------------------------------------------------------")
+    print("Open index.html to view coverage")
```

Comment:
1) Would be nice to suppress the messages that this spits out to make the example cleaner
2) It doesn't work in CI at the moment due to npm
3) Am I calling this correctly?

Reply:
> Would be nice to suppress the messages that this spits out to make the example cleaner
Job for later - easy to suppress messages, but harder to do it safely so that we don't miss messages if the build fails

> It doesn't work in CI at the moment due to npm
Makes sense but should be fairly easy to setup when I have the chance

> Am I calling this correctly?
Yep, but perhaps we should set default path here:
https://github.com/VyperCore/bucket/blob/main/bucket/rw/html.py#L20C1-L20C66
(when called from command line there is a default path already)